### ISSUE TITLE
Validation the model using 30% of the data set

### ISSUE THREAD

**Issue by omarmaree** [AlexeyAB/darknet#4344]
@AlexeyAB 
Hello Alex, I'm training the YOLO3.cfg model and edited as mentioned in the documentation with 3 classes. My data set was around 1100 images and the valid file valid=data/train.txt, However, the trained model gave me map with 99%, testing the model with new data showed me very bad results. That means overfitting has happened. Then, I increased my data set to 3400 images and used 1170 images for validation by putting in the obj file valid=data/valid.txt, and valid.txt contains the names of the validation data set. Is that correct to do ?? and what is the procedure that used to validate the YOLO model? is the cross-validation is used? and how can I solve the problem of overfitting? 
Thanks in advance

**Comment by lsd1994** [comment]
That's correct, you can have a try.

**Comment by AlexeyAB** [comment]
> Then, I increased my data set to 3400 images and used 1170 images for validation by putting in the obj file valid=data/valid.txt, and valid.txt contains the names of the validation data set. Is that correct to do ??

Yes. 
F.e. if you have 4000 images in dataset, you **must randomly** divide it to 3000 images in Training dataset and 1000 images in Validation dataset.


> and what is the procedure that used to validate the YOLO model? is the cross-validation is used? 

https://github.com/AlexeyAB/darknet#when-should-i-stop-training

> and how can I solve the problem of overfitting?

https://github.com/AlexeyAB/darknet#how-to-improve-object-detection

> for each object which you want to detect - there must be at least 1 similar object in the Training dataset with about the same: shape, side of object, relative size, angle of rotation, tilt, illumination. So desirable that your training dataset include images with objects at diffrent: scales, rotations, lightings, from different sides, on different backgrounds - you should preferably have 2000 different images for each class or more, and you should train 2000*classes iterations or more

**Comment by omarmaree** [comment]
It worked and gave good accuracy. Thanks Alex
